File: 1 - BackEnd/Vanilla/PHP/3 - lang/6 - conversiones.php
<?php

/* ##########==================########## */
/* ######===--- Conversiones ---===###### */
/* ##########==================########## */

$numero = 5;
$string = "55.55";

// Convertir un dato a (string).
$string = ((string) $numero);

// Convertir un dato a (int).
$numero = ((int) $string);

// Convertir un dato a (float).
$numero = ((float) $string);

// Convertir un dato a (double).
$numero = ((double) $string);

// Convertir un dato a (booleano).
$numero = ((bool) $string);

// Convertir un dato a (objeto).
$objeto = ((object) obtener_objeto());

/* ##########=======================########## */
/* ######===--- De array a string ---===###### */
/* ##########=======================########## */

$arr = array("Hola", true, 35);

/* La funcion (implode) permite convertir un 'array' 
a una cadena de texto: 

	--- El primer parametro recibe el separador de los elementos de la cadena.
	--- El segundo parametro es el 'array' el cual se ha de convertir. */
$str = implode(", ", $arr); // Da: Hola, true, 35.

$str = implode(" ", $arr); // Da: Hola true 35.

$str = implode("", $arr); // Da: Holatrue35.

// ------------------------------------------ //
// ------ Convertir variables a arrays ------ //
// ------------------------------------------ //

$variable_0 = "Brandon";
$variable_1 = null;
$variable_2 = true;

/* El metodo (compact) crea un nuevo array asociativo con todas las variables 
espeficicadas. */
$arr_variables = compact("variable_0", "variable_1", "variable_2");

print_r($arr_variables);
/*
	Array
	(
	    [variable_0] => Brandon
	    [variable_1] => 
	    [variable_2] => 1
	)
*/

/* ##########=======================########## */
/* ######===--- Convertir a array ---===###### */
/* ##########=======================########## */

# Variable con datos separados por '/'.
$fecha = "2021/5/13";
$fecha_1 = "2021-5-13";
$fecha_2 = "2021 del mes 5 y el dia 13";

/* Explode sirve para convertir un string en un array, 
en este caso se separan los datos del string por una diagonal, 
el array elimina la diagonal y separa las palabras en datos de tipo 
array. */
$newArray = explode("/", $fecha);

$newArray_1 = explode("-", $fecha_1);

$newArray_2 = explode(" ", $fecha_2);

/* ===--- Dividir array por partes ---=== */

$newArray_2 = explode(" ", $fecha_2, 2);
/* El string se va a dividir en dos partes. */
/* 
  2021,
  del mes 5 y el dia 13
*/

$newArray_2 = explode(" ", $fecha_2, 3);
/* El string se va a dividir en tres partes. */
/* 
  2021,
  del,
  mes 5 y el dia 13
*/

$newArray_2 = explode(" ", $fecha_2, 4);
/* El string se va a dividir en cuatro partes. */
/* 
  2021,
  del,
  mes,
  5 y el dia 13
*/

$newArray_2 = explode(" ", $fecha_1, -1);
/* Divide el string en un array menos el ultimo elemento. */
/* 
  2021,
  5
*/

$newArray_2 = explode(" ", $fecha_1, -2);
/* Divide el string en un array menos los ultimos elemento. */
/* 
  2021
*/

File: 1 - BackEnd/Vanilla/Python/1 - lang/6 - conversiones.py
# ##########==================##########
# ######===--- Conversiones ---===######
# ##########==================##########

# Convertir un dato a (string).
string = str(5)

# Convertir un dato a int.
numero = int(float("55.55")) # Primero convertir a float para asegurar la conversión correcta

# Convertir un dato a float.
numero = float("55.55")

# Convertir un dato a booleano.
booleano = bool(string)  # True si string no está vacío

# ##########=======================##########
# ######===--- De Lista a String ---===######
# ##########=======================##########

# Convertir lista a string
arr = ["Hola", "true", "35"]

# El método join convierte una lista a una cadena, usando el separador especificado.
str_1 = ", ".join(arr)  # "Hola, true, 35"
str_2 = " ".join(arr)  # "Hola true 35"
str_3 = "".join(arr)  # "Holatrue35"

# ##########=======================##########
# ######===--- De String a Lista ---===######
# ##########=======================##########

fecha = "2021/5/13"
fecha_1 = "2021-5-13"
fecha_2 = "2021 del mes 5 y el dia 13"

# split() convierte un string en una lista, usando el delimitador especificado.
new_array = fecha.split("/")  # ['2021', '5', '13']
new_array_1 = fecha_1.split("-")  # ['2021', '5', '13']
new_array_2 = fecha_2.split(" ")  # ['2021', 'del', 'mes', '5', 'y', 'el', 'dia', '13']

# Limitar el número de partes
new_array_2_2 = fecha_2.split(" ", 2)  # ['2021', 'del', 'mes 5 y el dia 13']
new_array_2_3 = fecha_2.split(" ", 3)  # ['2021', 'del', 'mes', '5 y el dia 13']

# ##########========================##########
# ######===--- Variables a Listas ---===######
# ##########========================##########

# Variables
variable_0 = "Brandon"
variable_1 = None
variable_2 = True

# Convertir variables a diccionario.
arr_variables = {
    "variable_0": variable_0,
    "variable_1": variable_1,
    "variable_2": variable_2
}

print(arr_variables)
# Imprime: {'variable_0': 'Brandon', 'variable_1': None, 'variable_2': True}